contracts/finance/andromeda-validator-staking/src/contract.rs

```diff
@@ -235,15 +242,22 @@ fn execute_claim(
 
     let res = Response::new()
         .add_message(DistributionMsg::SetWithdrawAddress {
-            address: recipient.to_string(),
+            address: recipient_address.to_string(),
         })
         .add_message(DistributionMsg::WithdrawDelegatorReward {
             validator: validator.to_string(),
         })
         .add_attribute("action", "validator-claim-reward")
-        .add_attribute("recipient", recipient)
+        .add_attribute("recipient", recipient_address)
         .add_attribute("validator", validator.to_string());
 
+    if let Some(recipient) = recipient {
+        if recipient.msg.is_some() {
+            res.clone()
+                .add_submessage(recipient.generate_direct_msg(&deps.as_ref(), info.funds)?);
```

Comment:
Should the amount being sent here not be the amount that is being claimed?

Reply:
The more I think about this the more difficult I think this implementation might be.

Reply:
> Should the amount being sent here not be the amount that is being claimed?

I think it is not the amount to be claimed. By setting withdraw address and validator, the reward will be drawn to the recipient address. In case the recipient contain Execute message requiring funds, we might need to send `Claim` message with fund so that the fund can be sent with the Execute msg(in recipient).

However, I think it is somewhat weird to send funds with `Claim` message. There is still possibility that some people might send fund by accident and that fund will be sent to the recipient. I think it is rather safe to use `nonpayable` function as a guard for claim function and remove above logic

Reply:
Should we have a discussion about that?

Reply:
Let's go ahead and undo it, we can come back to this in a future release. Keep it as `AndrAddr` for now.